src/Filtering/ImageFeature/AdditiveGaussianNoiseImageFilter/Code.cxx

```diff
@@ -0,0 +1,78 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#include "itkImage.h"
+#include "itkImageFileReader.h"
+#include "itkImageFileWriter.h"
+#include "itkAdditiveGaussianNoiseImageFilter.h"
+
+int main( int argc, char* argv[] )
+{
+  // Check for proper arguments, if not, explain usage.
+  if( argc != 5 )
+    {
+    std::cerr << "Usage: "<< std::endl;
+    std::cerr << argv[0];
+    std::cerr << " <InputFileName> <OutputFileName> [Mean] [Standard Deviation]";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+    }
+    // Initialize and assign user provided variables
+  const char * inputImage = argv[1];
```

Comment:
Can these be std::string?

Reply:
Yes, I will change them to std::string

Reply:
@hjmjohnson @thewtex sorry to chime in but I'm wondering whether the same criterion is used in ITK. If not, and if it should be used, we may need to open an issue. Thanks.